GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -115,11 +124,12 @@ private abstract class Tree : IDisposable
             private readonly CancellationTokenSequence _reloadCancellationTokenSequence = new CancellationTokenSequence();
             private bool _firstReloadNodesSinceModuleChanged = true;
 
-            public void Dispose()
-            {
-                Detached();
-                _reloadCancellationTokenSequence.Dispose();
-            }
+            // A flag to indicate whether the data is currently being filtered or not.
+            // This helps to reduce unnecessary treeview rebinds.
+            private bool _isCurrentlyFiltering;
+
+            // A flag to indicate whether the data is being filtered (e.g. Show Current Branch Only).
+            private protected static AsyncLocal<bool> IsApplyFiltering = new AsyncLocal<bool>();
```

Comment:
The identifier `IsApplyFiltering` is stranger English than mine.

Reply:
Aye, not sure what I was thinking there...